### PR TITLE
[CELEBORN-173] refactor minicluster and fix ut

### PR DESCRIPTION
**Pull request by RexXiong**
### What changes were proposed in this pull request?
refactor minicluster for ut, and fix failed ut

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by RexXiong** [comment]
@zwangsheng @waitinfuture

File: worker/src/test/scala/org/apache/celeborn/service/deploy/worker/storage/DeviceMonitorSuite.scala [hunk 1]
```diff
@@ -357,7 +357,8 @@ class DeviceMonitorSuite extends AnyFunSuite {
       device1.notifyObserversOnNonCriticalError(mountPoints1, DiskStatus.READ_OR_WRITE_FAILURE)
       device1.notifyObserversOnNonCriticalError(mountPoints1, DiskStatus.IO_HANG)
       val deviceMonitorMetrics =
-        workerSource.gauges().filter(_.name.startsWith("Device")).sortBy(_.name)
+        workerSource.gauges().filter(_.name.startsWith("Device_" + device1.deviceInfo.name))
```
Comment (RexXiong) [inline worker/src/test/scala/org/apache/celeborn/service/deploy/worker/storage/DeviceMonitorSuite.scala:360]: DeviceMonitorSuite ut not sucess but not cause check failed, may be affected by mvn process exit.

**Comment by pan3793** [comment]
Does it has conflicts on branch-0.2, I think 0.2 has the same issues.

**Comment by waitinfuture** [comment]
> Does it has conflicts on branch-0.2, I think 0.2 has the same issues.

Seems it do have some conflicts but not hard to solve. Could @RexXiong plz also patch this to branch-0.2?

**Comment by RexXiong** [comment]
> > Does it has conflicts on branch-0.2, I think 0.2 has the same issues.
> 
> Seems it do have some conflicts but not hard to solve. Could @RexXiong plz also patch this to branch-0.2?

Sure.

**Pull request merged**